hooks/hiero_customize_export_ui.py

```diff
@@ -0,0 +1,286 @@
+# Copyright (c) 2018 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+#  UI Hook
+# ===========================
+
+import sgtk
+
+from sgtk.platform.qt import QtGui
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class HieroCustomizeExportUI(HookBaseClass):
+    """
+    This class defines methods that can be used to customize the UI of the various
+    Shotgun-related exporters. Each processor has its own set of create/get/set
+    methods, allowing for customizable UI elements for each type of export.
+    """
+    def create_shot_processor_widget(self, parent_widget):
+        """
+        Builds and returns a custom widget to be embedded in the parent exporter.
+        If a custom widget is returned by this method, it will be added to the
+        parent exporter's layout.
+
+        Example Implementation:
+
+        .. code-block:: python
+
+            widget = QtGui.QGroupBox("My Custom Properties", parent_widget)
+            widget.setLayout(QtGui.QFormLayout())
+            return widget
+
+        :param parent_widget: The parent widget.
+
+        :returns: A custom widget.
+        """
+        return None
+
+    def get_shot_processor_ui_properties(self):
+        """
+        Gets a list of property dictionaries describing the custom properties
+        required by the custom widget. This method will only be run if the
+        associated create widget hook method returns a widget. The dictionaries
+        will be turned into property widgets by the app before being passed to
+        the associated set properties hook method. The order that the dictionaries
+        are returned by this method is maintained when they are passed to the
+        associated set hook method.
+
+        Example Return Data:
+
+        .. code-block:: python
+
+            [
+                dict(
+                    label=label_str,
+                    key=key_str,
+                    value=property_value,
+                    type=property_type # Usually == type(value)
+                    tooltip=tooltip_str,
+                ),
+            ]
+
+        Example Implementation:
+
+        .. code-block:: python
+
+            return [
+                dict(
+                    label="Create Cut:",
+                    key="sgCreateCut",
+                    value=True,
+                    type=bool,
```

Comment:
Yeah...I just tried to think of an example of where it might be different than type(value), and I can't come up with anything reasonable. I'll ditch it!